backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/database/repos/CardRepository.kt

```
@@ -32,4 +41,8 @@ object CardRepository {
             this.revoked = false
             this.codeType = codeType
         }
+
+    fun reactivate(card: CardEntity, totpSecret: ByteArray) {
```

Maybe `activate`?

---

We should probably check that CardEntity is actually a dynamic card?

---

hmm. but I think actual error handling should not happen in the CardRepository itself, but before that. Adding an if/else still won't hurt though 🤷🏼‍♀️

---

Hmmm depends :D I guess the repositories should also be responsible for the entities' coherence/validation.
They do the manipulation, so they should check whether that is an allowed manipulation I guess... The concepts / responsiblities are quite blurry in our setup though....

---

so you think if/else is enough of a check or should we actually throw an error?